coalib/results/Diff.py

```diff
@@ -8,25 +8,34 @@
 from coala_utils.decorators import enforce_signature, generate_eq
 
 
-@generate_eq('_file', 'modified', 'rename', 'delete')
+@generate_eq('_file', 'modified', 'rename', 'delete', 'create')
 class Diff:
     """
     A Diff result represents a difference for one file.
     """
 
-    def __init__(self, file_list, rename=False, delete=False):
+    def __init__(self, file_list=None,
+                 rename=False, delete=False, create=False):
         """
         Creates an empty diff for the given file.
 
         :param file_list: The original (unmodified) file as a list of its
                           lines.
         :param rename:    False or str containing new name of file.
         :param delete:    True if file is set to be deleted.
+        :param create:    False or str containing the file name of the file to
+                          be created.
```

Comment:
We can do better... If file list is provided, we don't give it to self._file, it will be an empty list, what we'll do is, instead call add lines implicitly, to the given diff. Sounds same?